File: Liquid.Core/Entities/ProductTransaction.cs
using Liquid.Core.Enums;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Liquid.Core.Entities
{
    public class ProductTransaction : BaseEntity
    {
        [Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int ProductTransactionId { get; set; }
        public string ProductOrderNumber { get; set; }

        [ForeignKey("Product")]
        public int ProductId { get; set; }

        [ForeignKey("Customer")]
        public int? CustomerId { get; set; }
        public int Quantity { get; set; }
        public decimal PurchasedPrice { get; set; }
        public ProductTransactionType Type { get; set; }
        public Product Product { get; set; }
        public Customer? Customer { get; set; }
    }
}

File: Liquid.UI/src/app/shared/models/product.transaction.ts
import { ProductTransactionType } from "./product.transaction.type";
import { IProduct } from "./product";
import { ICustomer } from "./customer";
import { IBaseEntity } from "./base.entity";
export interface IProductTransaction extends IBaseEntity {
    productTransactionId: number;
    productOrderNumber: string;
    productId: number;
    customerId: number | null;
    quantity: number;
    purchasedPrice: number;
    type: ProductTransactionType;
    product?: IProduct;
    customer?: ICustomer;
}
